<?xml version="1.0"?>
<!-- Note: this is a launch file to tune this component for different tasks. -->
<launch>
    <arg name="cmd_vel_out" default="/cmd_vel" />
    <group ns="mcr_navigation/direct_base_controller">
        <node pkg="mcr_direct_base_controller" type="direct_base_controller_coordinator_node" name="coordinator" output="screen">
            <remap from="~pose_error" to="coordinator/pose_error" />
            <remap from="~pose_error_monitor_event_in" to="pose_error_monitor/event_in" />

            <remap from="~twist_synchronizer_event_in" to="twist_synchronizer/event_in" />
            <remap from="~pose_2" to="/mcr_navigation/direct_base_controller/input_pose" />

            <remap from="~pose_monitor_feedback" to="pose_error_monitor/feedback" />
            <remap from="~twist" to="$(arg cmd_vel_out)" />

            <remap from="~pose_error" to="coordinator/pose_error" />

            <!-- the cycle rate of the node (in Hz) -->
            <param name="loop_rate" type="double" value="100.0"/>

            <!-- Maximum linear velocities in the axes X, Y and Z (in meters/second) -->
            <param name="angular_synchronization" type="bool" value="true"/>

            <!-- ############# Parameters for transform_to_pose_converter_node ########### -->
            <param name="wait_for_transform" type="double" value="0.1"/>
            <param name="reference_frame" type="str" value="odom"/>
            <param name="target_frame" type="str" value="base_link"/>
            <!-- ############# Parameters for transform_to_pose_converter_node ########### -->

            <!-- ############# Parameters for twist_controller ########### -->
            <!-- proportional gains for the Cartesian linear velocities-->
            <param name="p_gain_x" type="double" value="1.4"/>
            <param name="p_gain_y" type="double" value="1.4"/>
            <param name="p_gain_yaw" type="double" value="1.2"/>
            <!-- ############# Parameters for twist_controller ########### -->

            <!-- ############# Parameters for twist_limiter ########### -->
            <!-- Maximum linear velocities in the axes X, Y and Z (in meters/second) -->
            <param name="max_velocity_x" type="double" value="0.6"/>
            <param name="max_velocity_y" type="double" value="0.6"/>
            <param name="max_velocity_z" type="double" value="0.0"/>

            <!-- Maximum angular velocities around the axes X, Y and Z (in radians/second) -->
            <param name="max_velocity_roll" type="double" value="0.0"/>
            <param name="max_velocity_pitch" type="double" value="0.0"/>
            <param name="max_velocity_yaw" type="double" value="0.6"/>
            <!-- ############# Parameters for twist_limiter ########### -->

            <!-- ############# Parameters for pose_monitor ########### -->
            <param name="threshold_linear_x" type="double" value="0.005" />
            <param name="threshold_linear_y" type="double" value="0.005" />

        </node>
    </group>
</launch>
